airflow-core/src/airflow/api_fastapi/core_api/routes/public/auth.py

```diff
@@ -55,3 +55,20 @@ def logout(request: Request, next: None | str = None) -> RedirectResponse:
         logout_url = request.app.state.auth_manager.get_url_login()
 
     return RedirectResponse(logout_url)
+
+
+@auth_router.get(
+    "/refresh",
+    responses=create_openapi_http_exception_doc([status.HTTP_307_TEMPORARY_REDIRECT]),
+)
+def refresh(request: Request, next: None | str = None) -> RedirectResponse:
+    """Refresh the authentication token."""
+    refresh_url = request.app.state.auth_manager.get_url_refresh()
```

Comment:
`refresh_url` can be `None`. In that case, we should redirect to logout? This scenario is when a user gets an expired token, they are redirected here but the auth manager the environment is using does not support refresh token.

Reply:
Yes, updated the code. Indeed, we should fall back to logging out in that case. That is logical.